skyportal/handlers/api/annotation_services.py

```diff
@@ -122,35 +124,47 @@ def post(self, obj_id):
             author = self.associated_user_object
 
             catalog = data.pop("catalog", "gaiadr3.gaia_source")
-            radius_arcsec = data.pop("crossmatchRadius", cfg["cross_match.gaia.radius"])
+            radius_degrees = (
+                data.pop("crossmatchRadius", cfg["cross_match.gaia.radius"]) / 3600.0
+            )  # convert arcsec to degrees
             limmag = data.pop("crossmatchLimmag", cfg["cross_match.gaia.limmag"])
             num_matches = data.pop("crossmatchNumber", cfg["cross_match.gaia.number"])
             candidate_coord = SkyCoord(ra=obj.ra * u.deg, dec=obj.dec * u.deg)
 
-            gc = GaiaClass()
-            sub_context = gc.GAIA_MESSAGES
-            conn_handler = gc._TapPlus__getconnhandler()
-            response = conn_handler.execute_tapget(sub_context, verbose=False)
-
-            maintenance = False
-            for line in response:
-                if "Maintenance ongoing" in line.decode("utf-8"):
-                    maintenance = True
-                    break
-
-            if maintenance:
-                return self.error("Gaia server down! Please try again later.")
-
-            df = (
-                GaiaClass()
-                .cone_search(
-                    candidate_coord,
-                    radius=radius_arcsec * u.arcsec,
-                    table_name=catalog,
+            query_string = f"""
+                SELECT DISTANCE(
+                POINT('ICRS', ra, dec),
+                POINT('ICRS', {obj.ra}, {obj.dec})) AS
+                source_id, ra, dec, ref_epoch,
+                phot_g_mean_mag, phot_bp_mean_mag, phot_rp_mean_mag,
+                pm, pmra, pmdec, parallax, parallax_error, parallax_over_error,
+                ruwe
+                FROM {{main_db}}.gaia_source
+                WHERE 1=CONTAINS(
+                POINT('ICRS', ra, dec),
+                CIRCLE('ICRS', {obj.ra}, {obj.dec},
+                        {radius_degrees}))
+            """
+
+            with gaia as g:
+                try:
+                    df = g.query(query_string)
+                    df = df.to_pandas()
+                except Exception as e:
+                    return self.error(
+                        f"Error querying Gaia: {e}. Please try again later."
```

Comment:
Should we mention the object name or RA/DEC or something in the error string?

Reply:
I'm not sure I see how that would help the user? But I can add that anyways

Reply:
added